include/openssl/bio.h

```diff
@@ -156,12 +156,9 @@ extern "C" {
 # define BIO_FLAGS_IO_SPECIAL    0x04
 # define BIO_FLAGS_RWS (BIO_FLAGS_READ|BIO_FLAGS_WRITE|BIO_FLAGS_IO_SPECIAL)
 # define BIO_FLAGS_SHOULD_RETRY  0x08
-# ifndef BIO_FLAGS_UPLINK
-/*
- * "UPLINK" flag denotes file descriptors provided by application. It
- * defaults to 0, as most platforms don't require UPLINK interface.
- */
-#  define BIO_FLAGS_UPLINK        0
+# if OPENSSL_API_COMPAT < 0x10200000L
```

Comment:
This PR is marked for backport to 1.1.1, but we should not be making deprecation changes in the public headers in a stable release.

Reply:
True. So we just except the last commit from the cherry-pick?

Reply:
Does this actually fix any bugs in 1.1.1? Or is it purely for cleanup reasons? If the latter then I'd say don't do it in 1.1.1 at all.

Reply:
Mostly cleanup. yes.
However it is fragile to depend on including header files in a certain sequence.
It is even more fragile if master does not have this dependency while 1.1.1 has it.
BTW: Is API_COMPAT < 0x10200000L still correct, at that time I thought the next version would be 1.2.0,
but as we know now it is 3.0.0

Reply:
Oh, good point. Should be:

````
# if !OPENSSL_API_3
````
I'm still minded to make this master only.

Reply:
okay, then master only.